src/main/java/picard/sam/util/ReadNameParser.java

```diff
@@ -136,18 +138,19 @@ private boolean readLocationInformation(final String readName, final PhysicalLoc
     }
 
     public boolean addLocationInformation(final String readName, final PhysicalLocation loc){
-        final PhysicalLocation location = locationCache.getIfPresent(readName);
-        if (location == null) {
-            if (readLocationInformation(readName, loc)) {
-                locationCache.put(readName, loc);
+        if (readName != readNameStored) {
+            boolean b = readLocationInformation(readName, loc);
+            if (b) {
```

Comment:
no need for this temp variable, `b`, just put the read....() statement in the if(.)